### PR TITLE
Add arithmetical commands

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Targets #267

---

I know that this PR will start a shitstorm, even if I tell you:
Calm down before reviewing this! I didn't do that to piss you
off or something!

---

So, here is why I wrote these implementations the way I did:
1. The code in `do_integral_arithmetic()` would be duplicated
   four times when writing it the other way.
2. I didn't use macros. Macros would be the alternative to this,
   but I know that some of you guys hate large macro foo, and this
   would clearly end up beeing a big shitload of macro foo!
3. I wonder how this will get optimized and whether the compiler
   is intelligent enough to optimize this into single functions.
   Probably it is not, but maybe it does some crazy optimization
   here we don't know yet.

---

I'm not pissed off or something if this gets rejected, but I
would like to have a calm and premeditated discussion about it,
without bad words and the like!

### CODE REVIEW

File: src/command/arithmetical.c [hunk 1]
```diff
@@ -25,37 +25,156 @@
...
+    while (ws_value_get_type(&args->value) != WS_VALUE_TYPE_NONE) {
+        if (ws_value_get_type(&args->value) != WS_VALUE_TYPE_INT) {
+            return -EINVAL;
+        }
+
+        intmax_t a = ws_value_int_get(&base->int_);
+        intmax_t b = ws_value_int_get(&args->int_);
+
+        ws_value_int_set(&base->int_, operation(a, b));
+    }
```
Comment (neithernut) [inline src/command/arithmetical.c:124]: I think you meant to iterate here...

**Comment by neithernut** [comment]
This setup doesn't make sense for divisions. Divisions only make sense for _two_ arguments. For subtractions, it's a different matter, because they are often presented as sums of negative numbers.

---

**If** you feel that it makes sense to move iteration logic to another function (in a _fixed_ set of functions) , then _why_ didn't you just use an enumeration?
Franky, I think it makes sense to have a macro for iterations. If would have the obvious benefit that it may be used in other places too. In this case, a macro would be the more useful and elegant.
I'll write such a macro and push a PR.

File: src/command/arithmetical.c (hunk 1 above)
Comment (matthiasbeyer) [inline src/command/arithmetical.c:124]: Yep.

**Comment by matthiasbeyer** [comment]
I like your suggestion!

**Comment by matthiasbeyer** [comment]
Fixed. I implemented the div directly.

**Comment by neithernut** [comment]
What do you intend to do wthi this branch now? You can either completely rewrite it or dump it.

**Comment by matthiasbeyer** [comment]
I fixed the division, the rest should work. I will of coures rewrite the helper function to use the new macros.

**Pull request merged**